doc/litespeed-configuration.rst

```diff
@@ -0,0 +1,156 @@
+.. _litespeed configuration:
+
+LiteSpeed Configuration
+-----------------------
+
+Below you will find detailed LiteSpeed configuration recommendations for the
+features provided by this library.
+
+Preamble
+~~~~~~~~
+
+First of all, let's get one thing straight here: You'll find a lot of documentation
+and noise around LiteSpeed cache on the Internet, mostly involving plugins, specifically the
+Wordpress one. You **don't** need any plugin to benefit from LiteSpeed cache!
+As long as you follow the HTTP specification regarding the caching headers, you can use it as
+a general reverse proxy just like NGINX or Varnish.
+
+LiteSpeed comes in two different variants:
+
+* OpenLiteSpeed (OLS) - the open source product with less features
+* LiteSpeed Web Server (LSWS) - the enterprise version with more features and professional support
+
+The caching module implementations are different and thus have to be configured differently but they support the
+same set of features (except for OLS not supporting Edge Side Includes (ESI)).
+
+So before you start configuring the server, make sure you know which version of LiteSpeed you are using.
+
+.. note::
+
+    Any LiteSpeed setup works in a single node web server environment only. If you are targeting a multi
+    node setup you might want to consider switching to :ref:`Varnish <varnish configuration>` which has excellent
+    support for this already built-in in this library.
+
+
+Configuring OpenLiteSpeed
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+OLS does not support different caching settings depending on ``.htaccess`` settings and different paths.
+If you need that, you have to go with LSWS instead.
+Thus, OLS has to be configured as follows on server or vHost level::
+
+    module cache {
+      # This enables the public cache
+      enableCache                      1
+
+      # This disables the private cache
+      enablePrivateCache               0
+
+      # This enables the public cache
+      checkPublicCache                 1
+
+      # This disables the private cache
+      checkPrivateCache                0
+
+      # Also consider the query string in caches
+      qsCache                          1
+
+      # Disable checking for a cached entry if there's a cookie on the request
+      reqCookieCache                   0
```

Comment:
we don't support user context caching with litespeed, right?

Reply:
Right. User context caching requires some kind of preflight request to the app to find out the context which requires implementation in the proxy.